src/fsharp/FSharp.Core/async.fs

```diff
@@ -1293,35 +1293,54 @@ namespace Microsoft.FSharp.Control
                 | Choice1Of2 computations ->
                     ProtectedCode ctxt (fun ctxt ->
                         let ctxtWithSync = DelimitSyncContext ctxt
+                        let count = ref computations.Length
                         let noneCount = ref 0
-                        let exnCount = ref 0
+                        let someOrExnCount = ref 0
```

Comment:
Would be good to make these all `let mutable` bindings to avoid the extra ref cell allocations